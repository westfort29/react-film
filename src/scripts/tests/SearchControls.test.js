import React from 'react';
import { SearchControls } from '../';
import renderer from 'react-test-renderer';

const SEARCH_BY_OPTIONS = [
  { id:'1', name: '1' },
  { id:'2', name: '2' }
];

test('SearchControls should contain right data', () => {
  const component = renderer.create(
    <SearchControls
      onSearchClick={() => {}}
      searchInputRef={ React.createRef() }
      searchByOptions={ SEARCH_BY_OPTIONS }
      onSearchOptionClick={() => {}}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

