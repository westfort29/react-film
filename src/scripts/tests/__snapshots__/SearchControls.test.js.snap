// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchControls should contain right data 1`] = `
<section
  className="search-controls"
>
  <div
    className="search-controls__background-shading"
  />
  <section
    className="search-controls__content"
  >
    <h1
      className="search-controls__heading"
    >
      Find your movie
    </h1>
    <input
      className="search-controls__search-input"
      onKeyUp={[Function]}
      placeholder="Search"
      type="text"
    />
    <div
      className="search-controls__controls"
    >
      <ul
        className="search-controls__search-by-list"
      >
         Search by:
        <li
          className="search-controls__search-by-item"
        >
          <button
            className="search-controls__button search-controls__button--inactive"
            onClick={[Function]}
          >
            1
          </button>
        </li>
        <li
          className="search-controls__search-by-item"
        >
          <button
            className="search-controls__button search-controls__button--inactive"
            onClick={[Function]}
          >
            2
          </button>
        </li>
      </ul>
      <button
        className="search-controls__button search-controls__button--large"
        onClick={[Function]}
      >
         Search 
      </button>
    </div>
  </section>
</section>
`;
